File: learn-pcap4j/src/main/java/com.greatdreams.learn.pcap4j/MainProgram.java
package com.greatdreams.learn.pcap4j;

import com.greatdreams.learn.pcap4j.test.Pca4jLoop;
import org.pcap4j.core.NotOpenException;
import org.pcap4j.core.PcapNativeException;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import java.io.IOException;

public class MainProgram {
    private static Logger logger = LoggerFactory.getLogger(MainProgram.class);
    public static void main(String[] args) throws InterruptedException, PcapNativeException, NotOpenException, IOException {
        logger.info("Hello pcap4j, application begins to run...");
        Pca4jLoop.main(args);
        logger.info("The application exit!");
    }
}

File: learn-springbatch/src/main/kotlin/com/greatdreams/learn/springbatch/MainProgram.kt
package com.greatdreams.learn.springbatch

import com.greatdreams.learn.springbatch.config.ApplicationConfig
import org.slf4j.LoggerFactory
import org.springframework.batch.core.*
import org.springframework.batch.core.job.SimpleJob
import org.springframework.batch.core.repository.JobRepository
import org.springframework.batch.core.repository.JobRestartException
import org.springframework.batch.core.repository.support.SimpleJobRepository
import org.springframework.boot.SpringApplication
import org.springframework.boot.autoconfigure.EnableAutoConfiguration
import org.springframework.context.annotation.ComponentScan
import org.springframework.context.annotation.Import
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@EnableAutoConfiguration
@ComponentScan("com.greatdreams.learn.springbatch.")
object MainProgram {
    val logger = LoggerFactory.getLogger(MainProgram.javaClass)

    @RequestMapping("/")
    fun home() : String {
        return "Hello World"
    }

    @JvmStatic fun main(args : Array<String>) {
        logger.info("Welcome, springboot application!")
        val ctx = SpringApplication.run(MainProgram::class.java, *args)

        var i = 1
        ctx.beanDefinitionNames.forEach { beanName ->
            logger.debug("$i\t$beanName\t${ctx.getType(beanName).canonicalName}")
            i++
        }
    }
}
